sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/LookupCatalog.scala

```diff
@@ -133,7 +133,11 @@ private[sql] trait LookupCatalog extends Logging {
         // For example, if the name of a custom catalog is the same with `GLOBAL_TEMP_DATABASE`,
         // this custom catalog can't be accessed.
         if (nameParts.head.equalsIgnoreCase(globalTempDB)) {
```

Comment:
How about we follow the table loopup logic here
```
object CatalogAndIdentifierParts {
  def unapply(nameParts: Seq[String]): Option[(CatalogPlugin, Seq[String])] = {
    if (nameParts.length == 1) {
      Some((currentCatalog, currentNamespace ++ nameParts))
    } else if (nameParts.length == 2 && nameParts.head.equalsIgnoreCase(globalTempDB)) {
      Some((catalogManager.v2SessionCatalog, nameParts))
    } else {
      try ... catch ...
    }
  }
}
```

Reply:
good idea! I am also thinking combining `CatalogAndIdentifier` and `CatalogAndIdentifierParts` into one such that catalog resolution is consistent. For example, if you resolve `global_temp.t` with `CatalogAndIdentifier`, catalog can be resolved as a v2 catalog named `global_temp`.  I haven't tested if this regresses, but what do you think?

Reply:
Let's just change `CatalogAndIdentifierParts` first. We can have a followup to clean up these patterns defined in `LookupCatalog.scala`.

Reply:
> ``` 
> if (nameParts.length == 1) {
>   Some((currentCatalog, currentNamespace ++ nameParts))
> }
> ```
Look like we cannot use the table look up logic since we have a command like `SHOW TABLES FROM testcat` where `testcat` needs to be resolved as a catalog.

Basically, we have a conflict when one part is given:
1. It needs to be resolved as a catalog:
```
CREATE TABLE testcat.table (id bigint, data string) USING foo
SHOW TABLES FROM testcat
```
2. It needs to be resolved as a non-catalog multi parts:
```
CREATE TABLE testcat.testcat (id bigint, data string) USING foo
USE testcat
DESCRIBE TABLE testcat
```
One way to resolve the conflict is to check the one part name against the current catalog. If they are the same, the one part name is used as `Identifier` (case 2), otherwise, it is used as a catalog (case 1).  Does this sound reasonable?

Reply:
ah you are right. For a single-part name like `abc`, it may mean a table `abc` under default catalog, or mean catalog `abc`.

I think it's better know if it's allowed to return a single catalog. e.g. it's allowed in `SHOW TABLES`, but not `DESCRIBE TABLE`.

Shall we separate `CatalogAndIdentifierParts` into `CatalogAndTable`, `CatalogAndNamespace`? For `CatalogAndTable`, it's not allowed to return empty array as table name, so we shouldn't resolve single-part name to a catalog. For `CatalogAndNamespace`, root namespace's name is an empty array, so it's allowed.

Reply:
Cool! I think it's a clean solution.

Btw, I am not going to address using the current namespace in this PR. This may not be the right place because we shouldn't be putting current namespace if the intention was to get the view name. I will address this in a separate PR for [SPARK-30094](https://issues.apache.org/jira/browse/SPARK-30094)

Reply:
SGTM